src/com/google/javascript/jscomp/PeepholeFoldConstants.java

```diff
@@ -1064,15 +1064,22 @@ static TernaryValue evaluateComparison(int op, Node left, Node right, boolean us
 
       case Token.STRING:
         if (right.isString()) {
-          // Only eval if they are the same type
+          return compareStrings(op, left, right);
+        } else {
+          // See http://www.ecma-international.org/ecma-262/6.0/#sec-abstract-relational-comparison
           switch (op) {
-            case Token.SHEQ:
+            case Token.GE:
+            case Token.LE:
+            case Token.GT:
+            case Token.LT:
+              return compareAsNumbers(op, left, right, useTypes);
```

Comment:
I agree. I've done a large rewrite that is based on the ES6 specs. Hopefully the code structure is better...